component/src/main/java/org/wso2/siddhi/runner/beam/SiddhiAppContainer.java

```diff
@@ -0,0 +1,284 @@
+
+    public void createSiddhiQuery() {
+        log.debug("Creating Siddhi Query");
+        ExecutionContext context = ExecutionContext.getInstance();
+        this.graph = context.getGraph();
+        for (CommittedBundle rootBundle: context.getRootBundles()) {
+            for (AppliedPTransform transform: graph.getPerElementConsumers(rootBundle.getPCollection())) {
+                if (SiddhiAppContainer.compatibleTransform(transform)) {
+                    generateSiddhiQueryForTransform(transform, rootBundle.getPCollection());
+                } else {
+                    log.error("Siddhi does not support " + transform.getTransform().toString() + " at the moment");
+                }
+            }
+        }
+    }
+
+    public void createSiddhiRuntime() throws SiddhiParserException {
+        try {
+            log.debug("Creating Siddhi Runtime");
+            SiddhiManager siddhiManager = new SiddhiManager();
+            StringBuilder siddhiApp = new StringBuilder();
+            this.streamDefinitions.forEach(siddhiApp::append);
+            this.queryDefinitions.forEach(siddhiApp::append);
+            log.debug(siddhiApp.toString());
+            siddhiManager.setExtension("beam:pardo", BeamParDoProcessor.class);
+            siddhiManager.setExtension("beam:groupbykey", BeamGroupByKeyProcessor.class);
+            siddhiManager.setExtension("beam:sink", BeamSinkProcessor.class);
+            this.runtime = siddhiManager.createSiddhiAppRuntime(siddhiApp.toString());
+            this.runtime.start();
+        } catch (SiddhiParserException e) {
+            log.error("There is an error in your SiddhiQL", e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    private void generateSiddhiQueryForTransform(AppliedPTransform transform, PCollection keyCollection) {
+
+        //If transform is not in HashMap
+        if (this.transformsMap.get(SiddhiAppContainer.generateTransformName(transform.getFullName())) == null) {
+
+            // Add stream definition and to HashMap for given transform
+            String streamName = SiddhiAppContainer.generateTransformName(transform.getFullName()) + "Stream";
+            String stream = "define stream " + streamName + " (event object);";
+            this.streamDefinitions.add(stream);
+            this.transformsMap.put(SiddhiAppContainer.generateTransformName(transform.getFullName()), transform);
+            this.collectionsMap.put(SiddhiAppContainer.generateTransformName(transform.getFullName()), keyCollection);
+
+            //Create queries for each transform mapped by output collections
+            if (transform.getOutputs().isEmpty()) {
+                if (transform.getTransform() instanceof TextIO.Write) {
+                    String sinkType = "text";
+                    String sinkStreamName = "textSinkStream"
+                            + (multiSinkcount == 1 ? "" : String.valueOf(multiSinkcount));
+                    //TODO look into exception listener
```

Comment:
Please fix todo's

Reply:
This is a suggested improvement for exception handling. I'll remove this.